Sources/AIProxy/OpenAI/OpenAITextToSpeechRequestBody.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  
+//  OpenAITextToSpeechCodables.swift
+//
+//
+//  Created by Daniel Aditya Istyana on 10/9/24.
+//
+
+
+import Foundation
+
+public struct OpenAITextToSpeechRequestBody: Encodable {
+
+    // Required
+    /// The text to generate audio for. The maximum length is 4096 characters.
+    public let input: String
+
+    // Required
```

Comment:
I would like to make a couple changes here.

1. The pattern that I have been using is to separate sections with a single `// Required` comment followed by the required members in alphabetical order, then an `// Optional` comment followed by the optional members in alphabetical order. I understand that this isn't totally necessary, since the types should communicate what is required versions optional (T vs T?), but for consistency let's stick with that for now.


2. Can you add a docstring to the top of this type:

```
/// Docstrings from https://platform.openai.com/docs/api-reference/audio/createSpeech
public struct OpenAITextToSpeechRequestBody: Encodable {...}
```

3. Per the reference above, `responseFormat` and `speed` are both optional, can you make their types:

public let responseFormat: ResponseFormat?

public let speed: Float?

I realize that you have satisfied the optionality in the initializer, which is also nice! For consistency, though, I would prefer to have the member's type communicate the optionality.

4. OpenAI was the first service I added support for, and as such I hadn't landed on the style yet for the nested types. I have since started doing this with other services (you can see the Groq integration as an example), which I like a lot for navigating with ctrl-6.

```
public struct OpenAITextToSpeechRequestBody: Encodable {
  // ...
  public let responseFormat: ResponseFormat
  // ...
}

// MARK: - RequestBody.ResponseFormat
public extension OpenAITextToSpeechRequestBody {
  enum ResponseFormat: String, Encodable {
          case mp3, opus, aac, flac, wav, pcm
   }
}
```

For this type, it may not seem like this is an improvement! However, when a request body is deeply nested this flattening approach greatly improves legibility. Take a look at the Mark `RequestBody.Message.UserContent.ContentPart` in GroqChatCompletionRequestBody.swift to see what I mean. And now consider the request body if all of those types were nested in a single code block. I would like to stick with this pattern for consistency.

Reply:
Also for part 4, please do the same for `Voice` and `Model`

Reply:
In case the Ctrl-6 part was confusing. This is the output in Xcode if you use Ctrl-6 while navigating the Groq chat completion request body. I find this very nice for understanding the structure (better than many nested indents IMO) 
![Screenshot 2024-10-09 at 10 46 04](https://github.com/user-attachments/assets/a67330ee-904c-4982-89c5-629826448588)